### ISSUE TITLE
能否支持openai key翻译

### ISSUE THREAD

**Issue by heheda123123** [chunibyocola/sc-translator-crx#162]
如题，能否支持填入api地址和key就可以翻译。从翻译效果来说，chatgpt的效果确实好了很多

**Comment by chunibyocola** [comment]
这个比较难，最主要是因为我不使用 ChatGPT，所以也没有账号去测试之类的。开发完成后出现问题后的维护也比较麻烦，还是因为那一点，因为我不使用。

**Comment by heheda123123** [comment]
好的，这个看个人使用习惯哈。不过还是推荐大佬试试。
国内的话可以用一些代理api，充值使用啥的也很方便，不受代理的限制
![image](https://github.com/chunibyocola/sc-translator-crx/assets/97490782/a7e1613f-32e9-40e4-a0ea-b2b57327de8a)

**Comment by freedomspiritJohn** [comment]
要是加的话加上谷歌Gemini

**Comment by xlenco** [comment]
如果支持 Openai接口，那么不光ChatGPT可以使用，国内几乎所有的Ai模型都可以使用。
例子：https://platform.moonshot.cn/docs/guide/migrating-from-openai-to-kimi#%E5%85%B3%E4%BA%8E-api-%E5%85%BC%E5%AE%B9%E6%80%A7
类似沉浸式翻译这样的扩展很早就有了。

**Comment by kmahyyg** [comment]
openai 格式是 ai 模型调用的通用标准，就基本上只要加个接口 url 输入和 api key 输入，剩下的都不用管，nodejs 都有 sdk 的

**Comment by kmahyyg** [comment]
> 这个需求一直都有人在关注，虽然我还是没怎么用过这些 AI，但我会尝试一下能否不用 SDK 的方式去添加 OpenAI 及能兼容 OpenAI API 的源和 Gemini。

Yes, it's possible, please refer to https://platform.openai.com/docs/api-reference/chat . I personally suppose, utilizing axios should be enough.

**Comment by heheda123123** [comment]
> 这个需求一直都有人在关注，虽然我还是没怎么用过这些 AI，但我会尝试一下能否不用 SDK 的方式去添加 OpenAI 及能兼容 OpenAI API 的源和 Gemini。

现在有很多api中转网站, 他们有些来源价格非常便宜(比如逆向)
